addons/survey/static/src/js/survey_form.js

```diff
@@ -0,0 +1,272 @@
+    start: function () {
+        var self = this;
+        self.surveyToken = self.$target.data('surveyToken');
+        self.answerToken = self.$target.data('answerToken');
+
+        var $timer = $('.o_survey_timer');
+        if ($timer.length) {
+            self.surveyTimerWidget = new publicWidget.registry.SurveyTimerWidget(null, $timer.data());
+            self.surveyTimerWidget.attachTo($timer);
+            self.surveyTimerWidget.on('time_up', self, function (ev) {
+                self.$target.find('button[name="button_submit"]').click();
+            });
+        }
+
+        _.each(this.$('div.o_survey_form_date'), function (date_group) {
+            self._updateDateForDisplay($(date_group));
+        });
+
+        _.each(this.$('div.o_survey_form_simple_radio'), function (radio_group) {
+            self._updateSimpleRadioForDisplay($(radio_group));
+        });
+
+        _.each(this.$('div.o_survey_form_simple_select'), function (select_group) {
+            self._updateSimpleSelectForDisplay($(select_group));
+        });
+
+        return this._super.apply(this, arguments);
+    },
+
+    // -------------------------------------------------------------------------
+    // Private
+    // -------------------------------------------------------------------------
+
+    // EVENTS
+    // -------------------------------------------------------------------------
+    _onChangeRadio: function (event) {
+        var $radio_group = $(event.currentTarget).parents('.o_survey_form_simple_radio');
+        return this._updateSimpleRadioForDisplay($radio_group);
+    },
+
+    _onChangeSelect: function (event) {
+        var $select_group = $(event.currentTarget).parents('.o_survey_form_simple_select');
+        return this._updateSimpleSelectForDisplay($select_group);
+    },
+
+    _onChangeMultipleCheckbox: function (event) {
+        var $elem = $(event.currentTarget);
+        var $comment_input = $elem.parents('.o_survey_form_multiple').find('input[type="text"]');
+        if ($elem.prop('checked')) {
+            $comment_input.focus();
+        } else {
+            $comment_input.val('');
+        }
+    },
+
+    _onSubmit: function (event) {
+        var self = this;
+        event.preventDefault();
+        var $elem = $(event.currentTarget);
+        var submitValue = $elem.val();
+        var $form = $elem.parents('form.o_survey_form');
+        var formData = $form.serializeArray();
+        var params = {
+            button_submit: submitValue,
+        };
+        _.each(formData, function(data, idx){
+             params[data.name] = data.value;
+        })
+
+        this._resetErrors();
+
+        _.each($form.find('div.o_survey_form_date'), function (date_field) {
+            self._updateDateForSubmit($(date_field), params);
+        });
+
+        return rpc.query({
+            route: '/survey/validate/' + self.surveyToken + '/' + self.answerToken,
+            params: params,
+        }).then(function (result) {
```

Comment:
Maybe extract that part in a `_onSubmitValidateSuccess` method to split and ease reading / nesting.